materials/03-remote-hosting.qmd

```diff
@@ -2,4 +2,203 @@
+
+At the time, we chose to use HTTPS as our security protocol for connecting the two repositories.
+
+**SAY SOMETHING ABOUT HTTPS vs SSH**
+
+Go into the local `planets` repository, and run this command:
+
+```{bash}
+$ git remote add origin https://github.com/<yourusername>/planets.git
+```
+
+which is equivalent to
+
+```{bash}
+$ git remote add origin <paste copied URL here>
+```
+
+`origin` is a local name used to refer to the remote repository. It could be called anything, but `origin` is a convention that is often used by default in git and GitHub, so it’s helpful to stick with this unless there’s a reason not to.
+
+We can check that the command has worked by running `git remote -v`:
+
+```{bash}
+$ git remote -v
+```
+
+``` bash-output-box
+origin   https://github.com/<yourusername>/planets.git (fetch)
+origin   https://github.com/<youusername>/planets.git (push)
+```
+
+## **3.2 Push Local Changes to a Remote Repository**
+
+Now that authentication is setup, we can return to the remote. This command will push the changes from our local repository to the repository on GitHub:
+
+```{bash}
+$ git push origin main
+```
+
+``` bash-output-box
+Enumerating objects: 16, done.
+Counting objects: 100% (16/16), done.
+Delta compression using up to 8 threads.
+Compressing objects: 100% (11/11), done.
+Writing objects: 100% (16/16), 1.45 KiB | 372.00 KiB/s, done.
+Total 16 (delta 2), reused 0 (delta 0)
+remote: Resolving deltas: 100% (2/2), done.
+To https://github.com/<youusername>/planets.git
+ * [new branch]      main -> main
+```
+
+![](../materials/images/github-repo-after-first-push.svg)
+
+## **3.3 Ignoring Things**
+
+What if we have files that we do not want Git to track for us, like backup files created by our editor or intermediate files created during data analysis? This often is the case in BC Government data projects.
+
+Let's create a file called `a.csv` that contains some data about the Red Planet's average temperature. Open a new csv using Excel and add two columns named 'Time' and 'Temperature'. Add a few rows of data if you want and save it to the planets directory. Then check the output of git status.
+
+```{bash}
+$ git status
+```
+
+``` bash-output-box
+On branch main
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+
+    a.csv
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+Putting data files under version control is be a waste of disk space and may put privacy at risk. As many BC Stats projects involve data with PI or other sensitive information, we generally follow a best practice of **not pushing data to GitHub**. We accomplish this by creating a file in the root directory of our project called a **.gitignore**.
+
+Again, open up your favorite text editor and type the following line into the first line of the file:
+
+``` bash-output-box
+*.csv
+```
+
+Then save and close the file, naming it .gitignore. Don't forget the dot at the beginning!
```

Comment:
I think people won't be familiar with files called `.something` and often when you save it, it auto adds a .txt extension or something like that. Might need to make that clear here. (For example, I always have extensions showing in my file explorer so that I know whether I've named something `.gitignore` or `.gitignore.txt` by accident).

Reply:
Moved to Issues